File: shared/typescript/src/vehicle.ts
/**
 * Vehicle — public-facing subset of the `vehicles` table.
 * Mirrors ../contracts/vehicle.schema.json.
 */

import { z } from "zod";

export const StatusSchema = z.enum(["available", "sold", "hold"]);
export type Status = z.infer<typeof StatusSchema>;

export const ConditionSchema = z.enum(["new", "used", "cpo"]);
export type Condition = z.infer<typeof ConditionSchema>;

export const VehicleSchema = z
  .object({
    vin: z.string().length(17),
    make: z.string(),
    model: z.string(),
    year: z.number().int().min(1900).max(2100),
    trim: z.string().nullable().optional(),
    color_ext: z.string().nullable().optional(),
    color_int: z.string().nullable().optional(),
    mileage: z.number().int().min(0).nullable().optional(),
    price: z.number().min(0),
    condition: ConditionSchema.nullable().optional(),
    transmission: z.string().nullable().optional(),
    fuel_type: z.string().nullable().optional(),
    body_style: z.string().nullable().optional(),
    features: z.array(z.string()).default([]),
    status: StatusSchema,
    description: z.string().nullable().optional(),
  })
  .strict();

export type Vehicle = z.infer<typeof VehicleSchema>;

File: shared/python/src/voiceai_contracts/vehicle.py
"""Vehicle — public-facing subset of the `vehicles` table.

Mirrors shared/contracts/vehicle.schema.json.
Does not include internal fields like `id`, `embedding`, `created_at`, `updated_at`.
"""

from __future__ import annotations

from enum import Enum

from pydantic import BaseModel, ConfigDict, Field


class Status(str, Enum):
    AVAILABLE = "available"
    SOLD = "sold"
    HOLD = "hold"


class Condition(str, Enum):
    NEW = "new"
    USED = "used"
    CPO = "cpo"


class Vehicle(BaseModel):
    model_config = ConfigDict(extra="forbid")

    vin: str = Field(min_length=17, max_length=17)
    make: str
    model: str
    year: int = Field(ge=1900, le=2100)
    trim: str | None = None
    color_ext: str | None = None
    color_int: str | None = None
    mileage: int | None = Field(default=None, ge=0)
    price: float = Field(ge=0)
    condition: Condition | None = None
    transmission: str | None = None
    fuel_type: str | None = None
    body_style: str | None = None
    features: list[str] = Field(default_factory=list)
    status: Status
    description: str | None = None
